nullaway/src/main/java/com/uber/nullaway/handlers/Handler.java

```diff
@@ -273,14 +273,18 @@ void onDataflowVisitLambdaResultExpression(
       ExpressionTree tree, NullnessStore thenStore, NullnessStore elseStore);
 
   /**
-   * Called while creating the error message on a possible null/empty optional deference.
+   * It should return an error if any of the handlers detect an error in dereference. Currently,
+   * used by {@link OptionalEmptinessHandler} to give an error in case of get() call on potentially
+   * empty {@link java.util.Optional}.
    *
    * @param expr The AST node for the expression being matched.
+   * @param baseExpr The AST node for the base of dereference expression being matched.
    * @param state The current visitor state.
-   * @param errorMessage error message string and type of the error wrapped in {@link
-   *     com.uber.nullaway.NullAway.ErrorMessage}.
+   * @param analysis A reference to the running NullAway analysis.
+   * @return {@link ErrorMessage} if dereference causes some error, otherwise null
    */
-  void onPrepareErrorMessage(ExpressionTree expr, VisitorState state, ErrorMessage errorMessage);
+  ErrorMessage getErrorMessageInDereference(
```

Comment:
Can we rename to something like `checkDereferenceForError`?  For me, the name `getErrorMessageInDereference` suggests that there is already a known error, and this routine is just constructing the message.